python-client/giskard/push/perturbation.py

```diff
@@ -164,15 +168,28 @@ def _text(
     model,
     transformation_function,
     value_perturbed,
-):
+):          
     passed = False
     # Iterate over the possible text transformations
     for text_transformation in text_transfo_list:
         # Create the transformation
-        t = text_transformation(column=feature)
+        _is_typo_transformation = issubclass(text_transformation, TextTypoTransformation)
+        kwargs = {}
+        if _is_typo_transformation:
+            kwargs = {"rng_seed": 1241}
+
+        t = text_transformation(column=feature, **kwargs)
 
         # Transform the slice
-        transformed = ds_slice_copy.transform(t)
+        if _is_typo_transformation:
+            _hash = xxh3_128_hexdigest(
+                f"{', '.join(map(lambda x: repr(x), ds_slice_copy.df.values))}".encode("utf-8")
+            )
+            if _hash not in hashed_typo_transformations.keys():
+                hashed_typo_transformations.update({_hash: ds_slice_copy.transform(t)})
```

Comment:
this cache will never be emptied as long as the worker is running, can you make it a fixed-sized LRU cache?

Reply:
why only caching typo transformations and not the rest?

Reply:
> this cache will never be emptied as long as the worker is running, can you make it a fixed-sized LRU cache?

I think it's better to keep the cache as long as possible, it'll be confusing to use for instance `@lru_cache(maxsize=16)`, as user might loose a push notification about typo transformation at some point without knowing why. Is there a problem if the caches is never emptied?

> why only caching typo transformations and not the rest?

Among these, typo transformations are the only ones that are random
```py
    TextGenderTransformation,
    TextLowercase,
    TextPunctuationRemovalTransformation,
    TextTitleCase,
    TextTypoTransformation,
    TextUppercase,
```